src/ha-utils/ha/gcp/providers/GCPIAAS.go

```diff
@@ -0,0 +1,361 @@
+			if returnValue == false {
+				return 12
+			}
+		}
+	}
+	// If the load balancer doesn't exist, then create one.
+	if loadBalancer == nil {
+		log.Println("Load balancer with name ", loadBalancerName, "does not exist in GCP. Initiating creation of load balancer resource.")
+
+		var createLBInput gcpmodels.CreateLBInput = gcpmodels.CreateLBInput{}
+		var connDrain gcpmodels.ConnectionDrainInfo = gcpmodels.ConnectionDrainInfo{30}
+		var backend gcpmodels.Backend = gcpmodels.Backend{}
+		backend.BalancingMode = "CONNECTION"
+		backend.Group = vmGroup.SelfLink
+		
+		createLBInput.Name = loadBalancerName
+		createLBInput.LoadBalancingScheme = "INTERNAL"
+		createLBInput.Protocol = "TCP"
+		createLBInput.HealthChecks = []string{probe.SelfLink}
+		createLBInput.Backends = []gcpmodels.Backend{backend}
+		createLBInput.ConnectionDraining = connDrain
+		returnValue = iaasProvider.lbService.CreateLoadBalancer(createLBInput, regionName)
+		if returnValue == false {
+			return 13
+		}
+		loadBalancer, returnValue = iaasProvider.lbService.GetLoadBalancer(loadBalancerName, regionName)
+		if returnValue == false {
+			return 14
+		}
+	}
+
+	// Identify whether a load balancing rule exists with the given name.
+	lbRuleName = iaasProvider.getLoadBalancingRuleName()
+	lbRule, returnValue = iaasProvider.lbRuleService.GetLBRule(lbRuleName, regionName)
+	if returnValue == false {
+		log.Println("Failed to retrieve load balancing rule details - ", loadBalancerName)
+		return 15
+	} else if returnValue == true && lbRule != nil {
+		log.Println("Load balancing rule with name:", lbRuleName, "already exists in GCP.")
+	}
+	// If the load balancing rule doesn't exist, then create one.
+	if lbRule == nil {
+		log.Println("Load balancing rule with name ", lbRuleName, "does not exist in GCP. Initiating creation of load balancing rule resource.")
+
+		floatingIPNetwork, floatingIPSubNetwork, returnValue := iaasProvider.getNetworkInfo(iaasProvider.Config.SubnetName, regionName)
+		var createLBRuleInput gcpmodels.CreateLBRuleInput = gcpmodels.CreateLBRuleInput{}
+		createLBRuleInput.Name = lbRuleName
+		createLBRuleInput.IPAddress = iaasProvider.Config.FloatingIP
+		createLBRuleInput.IPProtocol = "TCP"
+		createLBRuleInput.Ports = []string{fmt.Sprintf("%d", iaasProvider.Config.SFBrokerPort), fmt.Sprintf("%d", iaasProvider.Config.SFReportPort), fmt.Sprintf("%d", iaasProvider.Config.SFExternalPort), fmt.Sprintf("%d", iaasProvider.Config.SFDephooksPort)}
+		createLBRuleInput.LoadBalancingScheme = "INTERNAL"
+		createLBRuleInput.Network = floatingIPNetwork
+		createLBRuleInput.Subnetwork = floatingIPSubNetwork
+		createLBRuleInput.BackendService = loadBalancer.SelfLink
+
+		returnValue = iaasProvider.lbRuleService.CreateLBRule(createLBRuleInput, regionName)
+		if returnValue == false {
+			return 16
+		}
+		loadBalancer, returnValue = iaasProvider.lbService.GetLoadBalancer(loadBalancerName, regionName)
+		if returnValue == false {
+			return 17
+		}
+	}
```

Comment:
method too long. Can you modularize it and move big chunks of logic into smaller functions with meaningful names which improves the readability of this big function?